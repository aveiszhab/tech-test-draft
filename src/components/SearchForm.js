import React, {useState} from 'react';
import '../styles/SearchForm.css';
import getImages from '../requests/getImages'
import PropTypes from 'prop-types';

const SearchForm = ({setSearchResultList}) => {

  const [searchText, setSearchText] = useState('');
  const [isLoading, setIsLoading] = useState(false);

  const handleInputChange = (e) => {
    setSearchText(e.target.value)
  };

  const handleSubmit = async (event) => {
    setIsLoading(true);
    event.preventDefault();
    setSearchResultList( await getImages(searchText));
    setIsLoading(false);
  }

  return (
    <>
      <form 
        className='search-form'
        onSubmit={handleSubmit}
      > 
          <input 
            className='search-input' 
            type='text'
            placeholder='Search for an image'
            onChange={handleInputChange}
          />
          
          <button 
            name='go'
            className='submit-button'
            type='submit'
          >
            Go!
          </button>    
      </form>
      {isLoading && <div className='loader'></div>}
      
    </>
  )
}

SearchForm.propTypes = {
  setSearchResultList: PropTypes.func.isRequired
};

export default SearchForm;