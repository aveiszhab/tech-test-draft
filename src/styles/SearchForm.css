.search-form {
    display: flex;
    flex-direction: row;
    justify-content: center;
    margin-bottom: 10%;
}

.search-input {
    width: 20vw;
    height: 2vw;
    border: 1px solid gray;
    border-radius: 4px 0px 0px 4px;
    padding: 2px;
    font-size: 2.5vh;
}

button {
    width: 5vw;
    border-radius: 0px 4px 4px 0px;
    background-color: rgb(228, 35, 35);
    border: none;
    font-size: 2.5vh;
    color: white;
    
}

.loader {
    border: 16px solid #f3f3f3; 
    border-top: 16px solid rgb(228, 35, 35);
    border-radius: 50%;
    width: 4vw;
    height: 4vw;
    animation: spin 2s linear infinite;
    align-self: center;
  }
  
  @keyframes spin {
    0% { transform: rotate(0deg); }
    100% { transform: rotate(360deg); }
  }