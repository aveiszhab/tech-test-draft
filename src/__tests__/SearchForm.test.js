import React from 'react';
import { render, screen,fireEvent, wait } from '@testing-library/react';
import SearchForm from '../components/SearchForm';

describe('Searchform', () => {
  it('renders correctly to match the SearchForm snapshot', () => {
    const mockHandleSubmit = jest.fn();
    const {asFragment} = render(
      <SearchForm setSearchResultList={mockHandleSubmit}/>
    );

    expect(asFragment()).toMatchSnapshot();
  });
  
  it('renders form correctly', () => {

    const mockHandleSubmit = jest.fn();
    render(<SearchForm setSearchResultList={mockHandleSubmit}/>);
    
    const textbox = screen.getByRole('textbox');

    expect(textbox).toHaveClass('search-input');
    expect(screen.getByPlaceholderText(/search/i)).toBeInTheDocument();
    
    const button = screen.getByRole('button', {name: /go/i})

    expect(button).toBeInTheDocument();
    expect(button).toBeEnabled();
    expect(button).toHaveClass('submit-button')
  });

  it('calls the onChange and onSubmit callback handler', async () => {
    
    const mockHandleSubmit = jest.fn();
    render(<SearchForm setSearchResultList={mockHandleSubmit}/>);
    
    const textbox = screen.getByRole('textbox');
    const button = screen.getByRole('button', {name: /go/i});

    fireEvent.change(textbox, {
      target: {value: 'moon'}
    });
    fireEvent.click(button);

    expect(textbox.value).toBe('moon');
    await wait(() => {
      expect(mockHandleSubmit).toHaveBeenCalledTimes(1);
    });
  });
});