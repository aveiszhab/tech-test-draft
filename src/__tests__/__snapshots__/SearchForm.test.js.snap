// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Searchform renders correctly to match the SearchForm snapshot 1`] = `
<DocumentFragment>
  <form
    class="search-form"
  >
    <input
      class="search-input"
      placeholder="Search for an image"
      type="text"
    />
    <button
      class="submit-button"
      name="go"
      type="submit"
    >
      Go!
    </button>
  </form>
</DocumentFragment>
`;
